ports/stm32/timer.c

```diff
@@ -546,25 +577,28 @@ STATIC mp_obj_t pyb_timer_init_helper(pyb_timer_obj_t *self, size_t n_args, cons
     // set the TIM configuration values
     TIM_Base_InitTypeDef *init = &self->tim.Init;
 
-    if (args[0].u_obj != mp_const_none) {
+    if (args[ARG_freq].u_obj != mp_const_none) {
         // set prescaler and period from desired frequency
-        init->Prescaler = compute_prescaler_period_from_freq(self, args[0].u_obj, &init->Period);
-    } else if (args[1].u_int != 0xffffffff && args[2].u_int != 0xffffffff) {
+        init->Prescaler = compute_prescaler_period_from_freq(self, args[ARG_freq].u_obj, &init->Period);
+    } else if (args[ARG_period].u_int != 0xffffffff && args[ARG_period_hz].u_int != 0xffffffff) {
+        // set prescaler and period from desired period and scaler
+        init->Prescaler = compute_prescaler_period_from_t(self, args[ARG_period].u_int, args[ARG_period_hz].u_int, &init->Period);
+    } else if (args[ARG_prescaler].u_int != 0xffffffff && args[ARG_period].u_int != 0xffffffff) {
         // set prescaler and period directly
-        init->Prescaler = args[1].u_int;
-        init->Period = args[2].u_int;
+        init->Prescaler = args[ARG_prescaler].u_int;
+        init->Period = args[ARG_period].u_int;
     } else {
         mp_raise_TypeError("must specify either freq, or prescaler and period");
```

Comment:
This message probably deserves rewording.

Reply:
Yes indeed, updated.